### PR TITLE
feat(RHTAPREL-628): add task to collect Pyxis params

### PR DESCRIPTION
**Pull request by johnbieren**
This commit adds a new task `collect-pyxis-params` to retrieve the pyxis server and secret from the data file. It then exposes those as results so that the pyxis tasks can use them, instead of relying on them to be passed as pipeline params. As a result, the rhtap-service-push and push-to-external-registry pipelines were modified to use this new task.

### CODE REVIEW

File: tasks/collect-pyxis-params/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+Tekton task that collects pyxis configuration options from the data file. The task looks at the data file
+in the workspace to extract the `server` and `secret` keys for Pyxis. These are both emitted as task results
+for downstream tasks to use.
+
+## Parameters
+
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
+| dataPath | Path to the JSON string of the merged data containing the pyxis configuration options to use | Yes | data.json |
```
Comment (mmalina) [inline tasks/collect-pyxis-params/README.md:11]: Nit: It's a JSON file, not a string. So maybe something like this?

"Path to the merged data JSON file generated by collect-data task and containing the pyxis configuration options to use"

File: tasks/collect-pyxis-params/collect-pyxis-params.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        #!/usr/bin/env sh
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.pyxis | has("server")' $DATA_FILE) == false ] ; then
```
Comment (mmalina) [inline tasks/collect-pyxis-params/collect-pyxis-params.yaml:44]: You can implement the default in `jq` directly: `jq -r '.pyxis.server // "production"' $DATA_FILE > $(results.server.path)`

File: tasks/collect-pyxis-params/collect-pyxis-params.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            echo "production" > $(results.server.path)
+        else
+            jq -r '.pyxis.server' $DATA_FILE > $(results.server.path)
+        fi
+        if [ $(jq '.pyxis | has("secret")' $DATA_FILE) == false ] ; then
+            echo "No secret provided in pyxis key. Failing..."
```
Comment (mmalina) [inline tasks/collect-pyxis-params/collect-pyxis-params.yaml:50]: What if there isn't even a `pyxis` key in the json? It will fail the same, but then this text will not be accurate. So I wonder if maybe we should add a check for that earlier? Or else you could at least modify this text to say: "pyxis key or secret missing in data JSON file."

File: tasks/collect-pyxis-params/tests/pre-apply-task-hook.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+# Create a dummy pyxis secret (and delete it first if it exists)
+kubectl delete secret test-collect-pyxis-params-cert --ignore-not-found
+kubectl create secret generic test-collect-pyxis-params-cert --from-literal=cert=mycert --from-literal=key=mykey
```
Comment (mmalina) [inline tasks/collect-pyxis-params/tests/pre-apply-task-hook.sh:5]: Why is this needed? The task doesn't actually do anything with the secret - that happens in the pyxis tasks.

File: tasks/collect-pyxis-params/README.md (hunk 1 above)
Comment (johnbieren) [reply tasks/collect-pyxis-params/README.md:11]: nice catch

File: tasks/collect-pyxis-params/tests/pre-apply-task-hook.sh (hunk 4 above)
Comment (johnbieren) [reply tasks/collect-pyxis-params/tests/pre-apply-task-hook.sh:5]: Good catch. This is leftover from when I was leaning in a different direction. Thanks

File: tasks/collect-pyxis-params/collect-pyxis-params.yaml (hunk 2 above)
Comment (johnbieren) [reply tasks/collect-pyxis-params/collect-pyxis-params.yaml]: much nicer. updated

File: tasks/collect-pyxis-params/collect-pyxis-params.yaml (hunk 3 above)
Comment (johnbieren) [reply tasks/collect-pyxis-params/collect-pyxis-params.yaml]: I updated the echo

File: tasks/collect-pyxis-params/collect-pyxis-params.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.pyxis | has("secret")' $DATA_FILE) == false ] ; then
+            echo "Pyxis key or secret missing in data JSON file"
+            exit 1
+        fi
+        jq -r '.pyxis.secret' $DATA_FILE > $(results.secret.path)
+        jq -r '.pyxis.server // "production"' $DATA_FILE > $(results.server.path)
```
Comment (davidmogar) [inline tasks/collect-pyxis-params/collect-pyxis-params.yaml:49]: Not sure if this should be here or in the Pyxis task. This could emit an empty result and the Pyxis one would decide the default. The only thing that we really have to pass is the secret, isn't it?

Comment (johnbieren) [reply tasks/collect-pyxis-params/collect-pyxis-params.yaml:49]: It could go there... I was trying to create a task to collect the things we currently pass as parameters and not have to change the tasks. That was my intent. The server could be determined in each of the pyxis tasks (as long as we ensure to keep them in sync)

Comment (davidmogar) [reply tasks/collect-pyxis-params/collect-pyxis-params.yaml:49]: Let's leave it like this and we rethink it in the future.

**Comment by davidmogar** [comment]
Could you improve the title? `add task to collect Pyxis params` or something like that? PR title and commit one.

**Comment by johnbieren** [comment]
> Could you improve the title? `add task to collect Pyxis params` or something like that? PR title and commit one.

done

**Review by davidmogar — APPROVED** [review]

**Review by mmalina — APPROVED** [review]
